dbtmetabase/parsers/dbt_manifest.py

```diff
@@ -201,12 +201,26 @@ def _read_model(
                 # would return the ref() written in the test, but if the model has an alias, that's not enough.
                 # It is better to use child['depends_on']['nodes'] and exclude the current model
 
-                depends_on_ids = set(child["depends_on"][model_type])
-                depends_on_ids.discard(model["unique_id"])
-                if not depends_on_ids:
+                # From experience, nodes contains at most two tables: the referenced model and the current model.
+                # Note, sometimes only the referenced model is returned.
+                depends_on_nodes = list(child["depends_on"][model_type])
+                if len(depends_on_nodes) > 2:
+                    logger().warning(f'Expected at most two nodes, got {len(depends_on_nodes)} nodes, skipping {model["unique_id"]}')
```

Comment:
Use lazy formatting (e.g. `%s`) instead of f-strings in logger statements.

Reply:
Sure will adjust. Can I ask why? Or is this just your convention?

Reply:
It's a common convention (not just mine) in many languages, my IDE even highlights it as a warning. 

As I understand it, string formatting is relatively expensive, so you want to make it as lazy as possible by only processing it whenever that statement is above the log level (e.g. if your log level is set to warning, you don't want to waste time formatting all the info statements). We don't have a high-performance code here by any means but just a convention.